### PR TITLE
Change size_t to std::size_t

### LINKED ISSUE #1844: Refactoring size_t to std::size_t
We have an inconsistent use of type `size_t` and `std::size_t`. 
Let's change everything to std::size_t.

### PR DESCRIPTION
**Pull request by darth-tytus**
Resolve #1844.

### CODE REVIEW

File: src/FunctorOps.cpp [hunk 1]
```diff
@@ -249,7 +249,7 @@ IntrinsicFunctors functorBuiltIn(std::string_view symbol, const std::vector<Type
     });
 }
 
-bool isValidFunctorOpArity(std::string_view symbol, size_t arity) {
+bool isValidFunctorOpArity(std::string_view symbol, std::size_t arity) {
```
Comment (TomasPuverle) [inline src/FunctorOps.cpp:252]: The files that use these should, strictly speaking, ensure to "#include <cstddef>"

File: src/ast/analysis/SCCGraph.h [hunk 2]
```diff
@@ -48,28 +48,28 @@ class SCCGraphAnalysis : public Analysis {
...
 
     /** Get the SCC of the given relation. */
-    size_t getSCC(const Relation* rel) const {
+    std::size_t getSCC(const Relation* rel) const {
         return relationToScc.at(rel);
     }
 
     /** Get all successor SCCs of a given SCC. */
-    const std::set<size_t>& getSuccessorSCCs(const size_t scc) const {
+    const std::set<std::size_t>& getSuccessorSCCs(const std::size_t scc) const {
```
Comment (TomasPuverle) [inline src/ast/analysis/SCCGraph.h:61]: I would suggest removing the "const" qualifiers.

**Review by TomasPuverle — COMMENTED** [review]
Nice one!
Minor nitpicks, LGTM.

**Review by b-scholz — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**